desktop/core/src/desktop/middleware.py

```diff
@@ -389,13 +390,19 @@ def __str__(self):
 
 class AuditLoggingMiddleware(object):
 
-  def __init__(self):
+  def __init__(self, get_response=None):
+    self.get_response = get_response
     self.impersonator = SERVER_USER.get()
 
     if not AUDIT_EVENT_LOG_DIR.get():
       LOG.info('Unloading AuditLoggingMiddleware')
       raise exceptions.MiddlewareNotUsed
 
+  def __call__(self, request):  
+    response = self.get_response(request)
+    self.process_response(request, response)
+    return response
+
   def process_response(self, request, response):
```

Comment:
`process_response` is now changed to https://docs.djangoproject.com/en/1.11/topics/http/middleware/
so either we need to subclass with `MiddlewareMixin` or just port it to the new format -->

```
class SimpleMiddleware:
    def __init__(self, get_response):
        self.get_response = get_response
        # One-time configuration and initialization.

    def __call__(self, request):
        # Code to be executed for each request before
        # the view (and later middleware) are called.

        response = self.get_response(request)

        # Code to be executed for each request/response after
        # the view is called.

        return response
```

Probably simpler to just to the later as compatible with Django 1/2? (`process_response` logic --> into `def __call__(..)`


https://docs.djangoproject.com/en/2.0/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware